### PR TITLE
STD MDF implementation

### PR DESCRIPTION
**Pull request by gospodnetic**
Issue #1262 
Implementation of equations, no unit tests yet.
Notes:
1) since equation for normal sampling uses arctan, I used the second version of function make_unit_vector which uses directly theta and phi.

2) Gamma function in disney code was implemented directly, as cmath already has the function, I used that one.

### CODE REVIEW

File: src/appleseed/foundation/math/microfacet.cpp [hunk 1]
```diff
@@ -773,4 +773,115 @@ float GTR1MDF::pdf(
...
+    const Vector3f&     v,
+    const Vector3f&     s,
+    const float         alpha_x,
+    const float         alpha_y,
+    const float         gamma) const
+{
+    const float phi = TwoPi<float>() * s[0];
+    const float a = gamma - 1.0f;
+    const float b = std::pow(1.0f - s[1], 1.0f / (1.0f - gamma)) - 1.0f;
+    const float theta = atan(alpha_x * std::sqrt(a * b));
```
Comment (est77) [inline src/appleseed/foundation/math/microfacet.cpp]: I think we need std:: here, std::atan(...)

File: src/appleseed/foundation/math/microfacet.h [hunk 2]
```diff
@@ -407,6 +407,61 @@ class GTR1MDF
...
+        const float         alpha_x,
+        const float         alpha_y,
+        const float         gamma) const APPLESEED_OVERRIDE;
+
+    virtual Vector3f sample(
+        const Vector3f&     v,
+        const Vector3f&     s,
+        const float         alpha_x,
+        const float         alpha_y,
+        const float         gamma = 0.0f) const APPLESEED_OVERRIDE;
```
Comment (est77) [inline src/appleseed/foundation/math/microfacet.h]: We can remove the default value for the gamma argument here.
Same thing in the following method declaration.

File: src/appleseed/foundation/math/microfacet.cpp (hunk 1 above)
Comment (gospodnetic) [inline src/appleseed/foundation/math/microfacet.cpp]: atan is a part of cmath which was already included. I'll change it to std:: - is it better to use std:: for trig. functs?

Comment (est77) [inline src/appleseed/foundation/math/microfacet.cpp]: This code used to be in a header (it was a template before). That's why all cmath functions are qualified. we could remove the std:: in all the file and add a using namespace std on top.
But that can be done in a different PR. For now we can just be consistent with the rest of the code in the file and use std:: qualifiers for cmath functions.

**Pull request merged**